pkg/dependency/parser/nodejs/npm/parse.go

```diff
@@ -208,7 +216,8 @@ func (p *Parser) resolveLinks(packages map[string]Package) {
 	}
 
 	workspaces := rootPkg.Workspaces
-	for pkgPath, pkg := range packages {
+	// Clone packages to avoid cases when we check already updated packages
```

Comment:
Could you elaborate on that?

Reply:
We update `packages` map inside of `packages` loop.
https://github.com/aquasecurity/trivy/blob/a2d1871387e511040e7eb8368d62c266b0197763/pkg/dependency/parser/nodejs/npm/parse.go#L232-L235
this is bad practice.

When I tested broken link with empty `resolved` field - this loop would endlessly update  `packages`, check them and  we get  `out of memory` error.

Reply:
Can we change the comment to something like "Changing the map during the map iteration causes unexpected behavior" for clarity?

Reply:
Yeah, it make sense.
Updated in https://github.com/aquasecurity/trivy/pull/6858/commits/b8e52c499fa94ebf59191fdcbf8b935e6584402a